polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -1029,3 +1033,125 @@ fn limit_and_sanitize_disputes<
 		(checked, checked_disputes_weight)
 	}
 }
+
+// Filters statements from disabled validators in `BackedCandidate` and `MultiDisputeStatementSet`.
+// Returns `true` if at least one statement is removed and `false` otherwise.
+fn filter_backed_statements<T: Config>(
+	backed_candidates: &mut Vec<BackedCandidate<<T as frame_system::Config>::Hash>>,
+) -> Result<bool, DispatchErrorWithPostInfo> {
+	let disabled_validators = shared::Pallet::<T>::disabled_validators();
+
+	if disabled_validators.len() == 0 {
+		// No disabled validators - nothing to do
+		return Ok(false)
+	}
+
+	// `BackedCandidate` contains `validator_indices` which are indecies within the validator group.
+	// To obtain `ValidatorIndex` from them we do the following steps:
+	// 1. Get `para_id` from `CandidateDescriptor`
+	// 2. Get the `core_idx` for the corresponding `para_id`
+	// 3. Get the validator group assigned to the corresponding core idx
```

Comment:
maybe it would be easier to go the other way and from disabled validators compute a bitmask of disabled indices within a group?

Reply:
Indeed it is :) Thanks for the suggestion!